timed/src/lib.rs

```diff
@@ -49,28 +49,64 @@ pub use timed_proc_macros::timed;
 
 #[macro_use]
 extern crate lazy_static;
+
 use std::collections::HashMap;
 use std::sync::Mutex;
+use std::time::Duration;
+use crate::Phase::Finish;
 
 lazy_static! {
     static ref TRACES: Mutex<HashMap<String, Vec<Hop>>> = Mutex::new(HashMap::new());
 }
 
-pub enum TracingStats {
-    None,
-    Percentage,
+#[derive(Copy, Clone)]
+pub struct TraceOptions {
+    pub statistics: Option<fn(&str)>,
+    pub chrome_trace: Option<fn(&str)>,
+}
+
+impl TraceOptions {
+    pub fn new() -> TraceOptions {
+        TraceOptions {
```

Comment:
I like this. I think this fits the builder pattern, I think you can straight away go ahead and use this https://docs.rs/derive_builder/0.9.0/derive_builder/

Also, that makes the trace_init() not necessary anymore.

Reply:
I'll change it, wasn't sure that you'll like it haha

Reply:
> Last thing is testing: can you think of a way we could test this? Can we somehow use the closures to tracing and statistics to write into a static variable which we can inspect afterwards? I guess it would also be enough to just test the TRACES static and ensure it contains the Hops, just unit test the statistics processing based on simple inputs rather than a whole program.

Definitely, and tbh I think the closures should receive a `ChromeTrace` / `Statistics` structs instead of `&str`s. This will cause the whole design to 'smell' more event driven, in the sense that the closures will get notified on each new `ChromeTrace` entry and will act upon it as they wish.
This will also make it far more testable, asserting each `ChromeTrace` in the closure.

I usually work on side projects on the weekends, so feel free to checkout this branch and implement it if you find some time.